rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/ai/TbAiNode.java

```diff
@@ -111,6 +127,12 @@ public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNode
             // LangChain4j AnthropicChatModel rejects requests with non-null ResponseFormat even if ResponseFormatType is TEXT
             responseFormat = config.getResponseFormat().toLangChainResponseFormat();
         }
+        if (config.getResourceIds() != null && !config.getResourceIds().isEmpty()) {
```

Comment:
Ensure that such resources actually exist and belong to the current tenant. You can fetch them using your resource data cache during initialization, perform the checks, and have the resources already cached since they will likely be requested when the AI node is used.